native_guide.md

```diff
@@ -0,0 +1,572 @@
+
+## Workflows <a name="workflows"></a>
+
+The design of codegen functionality provides mechanisms to run at different points in your app development lifecycle, including when you create or update an API as well as independently when you want to just update the data fetching requirements of your app but leave your API alone. It additionally allows you to work in a team where the schema is updated or managed by another person. Finally, you can also include the codgen in your build process so that it runs automatically (such as from in XCode).
+
+**Flow 1: Create API then automatically generate code**
+
+```bash
+$amplify init
+$amplify add api (select GraphQL)
+$amplify push
+```
+
+You’ll see questions as before, but now it will also automatically ask you if you want to generate GraphQL statements and do codegen. It will also respect the `./src/main` directory for Android projects. After the AppSync deployment finishes the Swift file will be automatically generated (Android you’ll need to kick off a [Gradle Build step](#androiduse)) and you can begin using in your app immediately.
+ 
+**Flow 2: Modify GraphQL schema, push, then automatically generate code**
+
+During development, you might wish to update your GraphQL schema a few times and generated code as part of an iterative dev/test cycle. Modify & save your schema in `./amplify/backend/api/<apiname>/schema.graphql` then run:
+
+```bash
+$amplify push
+```
+
+Each time you will be prompted to update the code in your API and also ask you if you want to run codegen again as well, including regeneration of the GraphQL statements from the new schema.
+ 
+**Flow 3: No API changes, just update GraphQL statements & generate code**
+
+One of the benefits of GraphQL is the client can define it's data fetching requirements independently of the API. Amplify codegen supports this by allowing you to modify the selection set (e.g. add/remove fields inside the curly braces) for the GraphQL statements and running type generation again. This gives you fine grained control over the network requests that your application is making. Modify your GraphQL statements (default in the `./graphql` folder unless you changed it) then save the files and run:
+
+```bash
+$amplify codegen types
+```
+A new updated Swift file will be created (or run Gradle Build on Android for the same). You can then use the updates in your application code.
+ 
+**Flow 4: Shared schema, modified elsewhere (e.g. console or team workflows)**
+
+Suppose you are working in a team and the schema is updated either from the AWS AppSync console or on another system. Your types are not out of date because your GraphQL statement was generated off an outdated schema. The easiest way to resolve this is to regenerate your GraphQL statements, update them if necessary, and then generate your types again. Modify the schema in the console or on a separate system, then run:
```

Comment:
"not out of date" should be "now out of date"

Reply:
Fixed